tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Controllers/OrdersControllerTests.cs

```diff
@@ -144,6 +144,242 @@ public async Task GetOrderSummaryAsync_IsSummaryComplete_ReturnResult()
             orderSummary.Should().BeEquivalentTo(expected);
         }
 
+        [Test]
+        public async Task GetOrderSummaryAsync_OrderPartyOrganisationNameMissing_ReturnsOrderPartyIncomplete()
+        {
+            const string orderId = "C0000014-01";
+            var context = OrdersControllerTestContext.Setup();
+
+            (Order order, OrderSummaryModel expected) = CreateOrderSummaryTestData(orderId, "Some Description", context.PrimaryOrganisationId);

```

Comment:
A lot of these tests are the same. I wonder if we can utilise `TestCaseSource`. See example found in ApplicationUserValidatorTests in Identitiy.Api or maybe @bjss-chris has a good idea?

Reply:
One option might be to use an expression tree, like this for example:

```c#
[Test]
public async Task GetOrderSummaryAsync_OrderPartyOrganisationNameMissing_ReturnsOrderPartyIncomplete()
{
	await GetOrderSummaryAsync_PropertyMissing_ReturnsOrderPartyIncomplete(o => o.OrganisationName);
}

private static async Task GetOrderSummaryAsync_PropertyMissing_ReturnsOrderPartyIncomplete<T>(Expression<Func<Order, T>> selector)
{
	const string orderId = "C0000014-01";
	var context = OrdersControllerTestContext.Setup();

	(Order order, OrderSummaryModel expected) = CreateOrderSummaryTestData(orderId, "Some Description", context.PrimaryOrganisationId);

	var orderParameter = selector.Parameters.First();
	var assignDefaultExpression = Expression.Lambda<Action<Order>>(
		Expression.Assign(selector.Body, Expression.Constant(default(T), typeof(T))), orderParameter);

	var assignDefault = assignDefaultExpression.Compile();
	assignDefault(order);

	expected.Sections.Single(s => s.Id == "ordering-party").Status = "incomplete";

	context.Order = order;

	using var controller = context.OrdersController;

	var result = await controller.GetOrderSummaryAsync(orderId) as OkObjectResult;
	var orderSummary = result.Value as OrderSummaryModel;
	orderSummary.Should().BeEquivalentTo(expected);
}
```
However, I'm normally a proponent for keeping tests as simple as possible and my concern with this approach is that it's a bit 'too clever' for a test, so I'm wondering if `TestCaseSource` as suggested by @sam-roberts-bjss might be a better fit...

Reply:
Switched to using Sams TestSource and removed individual field tests, Completion logic now only use OrganisationContact !=null to decide if OrderParty is complete.